airflow/contrib/operators/mysql_to_gcs.py

```diff
@@ -229,4 +233,9 @@ def type_map(cls, mysql_type):
             FIELD_TYPE.TIMESTAMP: 'TIMESTAMP',
             FIELD_TYPE.YEAR: 'INTEGER',
         }
-        return d[mysql_type] if mysql_type in d else 'STRING'
+        if mysql_type in d:
+            return d[mysql_type]
+        elif mysql_type_flag & FLAG.BINARY > 0:
```

Comment:
This is definitely an enhancement (instead of defaulting everything to STRING)!

The only subtle caveat here is that a Char/Varchar field with binary collation will have `field_flag & FLG.BINARY = 128`, even if the the field itself is not binary. 

You can provide an option in the operator to pass in mysql schema so it can override the field types determined here.